R/helpers.R

```diff
@@ -13,6 +13,46 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+.geocode <- function(name, country, method = "osm") {
+  if (missing(name) || !is.character(name)) {
+    stop("Please provide a valid location name as a string.")
+  }
+  
+  # Use match.arg to validate the method parameter
+  method <- match.arg(method, choices = c("osm", "census", "arcgis", "census_simple", "geocodio",
+                                          "mapbox", "google", "bing", "here", "tomtom", "nominatim", "tiger"))
+  
+  # Handle API keys for methods that require them
+  methods_with_keys <- c("google", "bing", "here", "tomtom", "mapbox", "geocodio")
+  if (method %in% methods_with_keys) {
+    api_key_env <- paste0(toupper(method), "_API_KEY")
+    api_key <- Sys.getenv(api_key_env)
+    if (api_key == "") {
+      stop(paste0("API key for ", method, " is required. Please set the '", api_key_env, "' environment variable."))
+    }
+  }
+  
+  result <- tryCatch({
+    geocode_df <- tibble::tibble(address = paste0(name, " ", country)) %>%
```

Comment:
Can you still get rid of these dependencies via `tidygeocoder::geocode(.tbl = data.frame(address=location), "address", method = method, quiet = TRUE)`, you probably also need `address <- NULL` at the top of the function to silence the `geocode: no visible binding for global variable ‘address’` note in https://github.com/koenderks/rcityviews/actions/runs/11346897862/job/31557368996#step:9:117

Reply:
Sure !